File: TMessagesProj/src/main/java/ua/gomin/messenger/configs/GominExperimentalConfig.java
package ua.gomin.messenger.configs;

import android.content.Context;
import android.content.SharedPreferences;

public class GominExperimentalConfig {

    public static final GominExperimentalConfig INSTANCE = new GominExperimentalConfig();
    private static final String PREFS_NAME = "gomin_experimental_config";

    

    private SharedPreferences getPrefs(Context context) {
        return context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }

    public boolean getTranslatorEnabled(Context context) {
        return getPrefs(context).getBoolean("translatorEnabled", false);
    }
    public boolean getJsonViewerEnabled(Context context) {
        return getPrefs(context).getBoolean("jsonViewerEnabled", false);
    }
    public void setTranslatorEnabled(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("translatorEnabled", value).apply();
    }
    public void setJsonViewerEnabled(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("jsonViewerEnabled", value).apply();
    }
}

File: TMessagesProj/src/main/java/ua/gomin/messenger/configs/GominExperimentalConfig.kt
package ua.gomin.messenger.configs

import android.content.Context
import android.content.SharedPreferences

/**
 * Експериментальні налаштування.
 */
object GominExperimentalConfig {

    private const val PREFS_NAME = "gomin_experimental_config"

    private fun getPrefs(context: Context): SharedPreferences {
        return context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }

    fun getTranslatorEnabled(context: Context): Boolean =
        getPrefs(context).getBoolean("translatorEnabled", false)

    fun setTranslatorEnabled(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("translatorEnabled", enabled).apply()
    }

    fun getJsonViewerEnabled(context: Context): Boolean =
        getPrefs(context).getBoolean("jsonViewerEnabled", false)

    fun setJsonViewerEnabled(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("jsonViewerEnabled", enabled).apply()
    }
}
